samcli/commands/local/invoke/cli.py

```diff
@@ -31,11 +34,10 @@
     "--event",
     "-e",
     type=click.Path(),
-    default=STDIN_FILE_NAME,  # Defaults to stdin
     help="JSON file containing event data passed to the Lambda function during invoke. If this option "
-    "is not specified, we will default to reading JSON from stdin",
+    "is not specified, no event is assumed. Pass in the value '-' to input JSON via stdin",
 )
-@click.option("--no-event", is_flag=True, default=False, help="Invoke Function with an empty event")
+@click.option("--no-event", is_flag=True, default=True, help="DEPRECATED: By default no event is assumed.", hidden=True)
```

Comment:
Do we have a plan to remove this option all together?

Reply:
I'm open to that, question is if it's worth breaking people who are already using it. But by making it hidden, we're more or less de facto deleting it. The next step would be to drop values on the floor if someone goes to the trouble of doing `--no-no-event` or similar.

Reply:
Can we actually add a line somewhere "DEPRECATED" onto stdout, before we start looking for values on stdin. Similair to how python2.7 was deprecated.

Reply:
You don't have `--no-no-event` defined. So customers shouldn't be doing that at least. (https://click.palletsprojects.com/en/7.x/options/#boolean-flags). 

I am ok not deleting it fully. You marked it as `DEPRECATED` in the help text, which would indicate we would remove it at some point. Maybe the right thing to do is not to that at all, and allow people to explicitly define `--no-event`. Once this is released, I am not sure why people would type more but it really doesn't hurt anything. If we are planning on removing completely, I would like us to have a way to do that was all. This is a non-blocking comment but worth thinking about.

Reply:
I think if someone explicitly defines `--no-event` there is no harm done - we're basically making it a no-op regardless, and I believe the deprecation flag removes it from help text anyways.

So perhaps the best way forward is to just leave it in place, but have it rendered inert.